UnitystationLauncher/ViewModels/SecurityPanelViewModel.cs

```diff
@@ -0,0 +1,538 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Reactive.Concurrency;
+using System.Threading.Tasks;
+using MessageBox.Avalonia.BaseWindows.Base;
+using ReactiveUI;
+using Serilog;
+using UnitystationLauncher.Infrastructure;
+using UnitystationLauncher.Models.Enums;
+using UnitystationLauncher.Services.Interface;
+
+namespace UnitystationLauncher.ViewModels;
+
+public class PipeHubBuildCommunication : IDisposable
+{
+    private NamedPipeServerStream _serverPipe;
+    private StreamReader? _reader;
+    private StreamWriter? _writer;
+
+    public PipeHubBuildCommunication()
+    {
+        _serverPipe = new NamedPipeServerStream("Unitystation_Hub_Build_Communication", PipeDirection.InOut, 1,
+            PipeTransmissionMode.Byte, PipeOptions.Asynchronous);
+    }
+
+    private enum ClientRequest
+    {
+        URL = 1,
+        API_URL = 2,
+        Host_Trust_Mode = 3,
+    }
+
+    public async void StartServerPipe()
```

Comment:
```csharp
public async Task StartServerPipe()
```

This should probably also be a service instead of in a ViewModel. This is going to be extremely hard to write tests for.

Reply:
ok, yeah I'll start pulling this out

Reply:
yeah idk since there is only one service method and that's start, 
shall I make it into a proper service though idk how to initialise it